pyiron_base/job/generic.py

```diff
@@ -159,8 +160,16 @@ def __init__(self, project, job_name):
         self._server = Server()
         self._logger = state.logger
         self._executable = None
-        self._status = JobStatus(db=project.db, job_id=self.job_id)
-        self.refresh_job_status()
+        if not state.database.database_is_disabled:
+            self._status = JobStatus(db=project.db, job_id=self.job_id)
+            self.refresh_job_status()
+        elif os.path.exists(self.project_hdf5.file_name):
+            initial_status = h5io.read_hdf5(
+                self.project_hdf5.file_name, job_name + "/status"
+            )
+            self._status = JobStatus(initial_status=initial_status)
+        else:
+            self._status = JobStatus()
```

Comment:
If possible all this logic should live inside the `JobStatus`. The `GenericJob` itself doesn't really care whether it's running with or without database, it just want's to have a status, I think it should be up to the `JobStatus` class where to retrieve that from.